torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -8821,14 +8821,63 @@ def celu_hook(module, _input, output):
             self.assertEqual(ddp_grads[0], local_model.fc.weight.grad)
             self.assertEqual(ddp_grads[1], local_model.fc.bias.grad)
 
-        def _test_hook_pickling(self, hook, hook_state):
+        def _test_hook_pickling_logs(
+            self,
+            hook,
+            hook_state,
+            restore_msg,
+        ):
+            chkpt_file = tempfile.gettempdir() + "/checkpoint.pt"
+            ddp_model = DistributedDataParallel(
+                torch.nn.Linear(1, 5).to(self.rank),
+                device_ids=[self.rank]
+            )
+            ddp_model.register_comm_hook(hook_state, hook)
+            state = {
+                'state_dict': ddp_model.state_dict(),
+                'comm_hook': hook,
+                'comm_hook_state': hook_state,
+            }
+
+            if self.rank == 0:
+                with self.assertLogs(level='WARNING') as captured:
+                    torch.save(state, chkpt_file)
+
+                # Check that the logger has only one entry
+                self.assertEqual(len(captured.records), 1)
+                # Check that the logger has an expected entry
+                self.assertRegex(
+                    captured.records[0].getMessage(),
+                    r'^NOTE: Process group .+ not serializable and excluded from a saved state\.$'
+                )
+
+            dist.barrier()
+            map_location = {'cuda:%d' % 0: 'cuda:%d' % self.rank}
+            with self.assertLogs(level='WARNING') as captured:
+                _ = torch.load(chkpt_file, map_location=map_location)
+
+            # Check that the logger has only one entry
+            self.assertEqual(len(captured.records), 1)
+
+            # Check that the logger has an expected entry
+            self.assertEqual(captured.records[0].getMessage(), restore_msg)
+
+            if self.rank == 0:
+                os.remove(chkpt_file)
+
+        def _test_hook_pickling_parity(
+            self,
+            hook,
+            hook_state,
+            special_attrs,
```

Comment:
Should this be passed in? Maybe, we just define this inside the method like `special_attrs = ["process_group", "rng"]` since we only ever pass those two special attributes and since this `_test_hook_pickling_parity()` has hard-coded logic specific to those two attributes (L8944 and L8961).